common/rulesys.cpp

```diff
@@ -24,462 +24,451 @@
 			break;
 		case BoolRule:
-			uint32 val = 0;
-			if (!strcasecmp(rule_value, "on") || !strcasecmp(rule_value, "true") || !strcasecmp(rule_value, "yes") || !strcasecmp(rule_value, "enabled") || !strcmp(rule_value, "1"))
-				val = 1;
-
-			m_RuleBoolValues[index] = val;
+			m_RuleBoolValues[index] = static_cast<uint32>(Strings::ToBool(rule_value));
 			LogRules("Set rule [{}] to value [{}]", rule_name, m_RuleBoolValues[index] == 1 ? "true" : "false");
 			break;
 	}
 
-	if(db_save)
-		_SaveRule(database, type, index);
+	if (db_save) {
+		_SaveRule(db, type, index);
+	}
 
-	return(true);
+	return true;
 }
 
 void RuleManager::ResetRules(bool reload) {
-	std::string expansion1;
-	std::string expansion2;
+	std::string client_rule;
+	std::string world_rule;
 
 	// these rules must not change during server runtime
 	if (reload) {
-		GetRule("World:ExpansionSettings", expansion1);
-		GetRule("World:UseClientBasedExpansionSettings", expansion2);
+		GetRule("World:UseClientBasedExpansionSettings", client_rule);
+		GetRule("World:ExpansionSettings", world_rule);
 	}
 
-	Log(Logs::Detail, Logs::Rules, "Resetting running rules to default values");
-	#define RULE_INT(cat, rule, default_value, notes) \
-		m_RuleIntValues[ Int__##rule ] = default_value;
-	#define RULE_REAL(cat, rule, default_value, notes) \
-		m_RuleRealValues[ Real__##rule ] = default_value;
-	#define RULE_BOOL(cat, rule, default_value, notes) \
-		m_RuleBoolValues[ Bool__##rule ] = default_value;
+	LogRulesDetail("Resetting running rules to default values.");
+
+	#define RULE_INT(category_name, rule_name, default_value, notes) \
+		m_RuleIntValues[ Int__##rule_name ] = default_value;
+	#define RULE_REAL(category_name, rule_name, default_value, notes) \
+		m_RuleRealValues[ Real__##rule_name ] = default_value;
+	#define RULE_BOOL(category_name, rule_name, default_value, notes) \
+		m_RuleBoolValues[ Bool__##rule_name ] = default_value;
 	#include "ruletypes.h"
 
 	// restore these rules to their pre-reset values
 	if (reload) {
-		SetRule("World:ExpansionSettings", expansion1.c_str(), nullptr, false, false);
-		SetRule("World:UseClientBasedExpansionSettings", expansion2.c_str(), nullptr, false, false);
+		SetRule("World:UseClientBasedExpansionSettings", client_rule);
+		SetRule("World:ExpansionSettings", world_rule);
 	}
 }
 
-bool RuleManager::_FindRule(const char *rule_name, RuleType &type_into, uint16 &index_into) {
-	if (rule_name == nullptr)
-		return(false);
+bool RuleManager::_FindRule(std::string rule_name, RuleType &type_into, uint16 &index_into) {
+	if (rule_name.empty()) {
+		return false;
+	}
 
-	int i;
-	int rule_count = CountRules();
-	for (i = 0; i < rule_count; i++) {
-		const RuleInfo &rule = s_RuleInfo[i];
-		if (strcmp(rule_name, rule.name) == 0) {
-			type_into = rule.type;
-			index_into = rule.rule_index;
-			return(true);
+	for (int i = 0; i < CountRules(); i++) {
+		const auto& r = s_RuleInfo[i];
+		if (Strings::Contains(rule_name, r.name)) {
```

Comment:
This changes behavior

We should have a `Strings::Equals` since `strcmp` is checking for exact equality of strings. We'll see side affects with partials here

Reply:
I'll add `Strings::Equals`.